[BUG][Task-API]load codec class fail

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

When using codec-related classes, the NoClassDefFoundError is caused due to the exclusion of related packages

ERROR] 2021-09-29 12:18:56.119 org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread:[216] - task scheduler failure
java.lang.NoClassDefFoundError: org/apache/commons/codec/binary/Base64
        at org.apache.dolphinscheduler.plugin.task.datasource.PasswordUtils.<clinit>(PasswordUtils.java:38)
        at org.apache.dolphinscheduler.plugin.task.datasource.mysql.MysqlDatasourceProcessor.getConnection(MysqlDatasourceProcessor.java:126)
        at org.apache.dolphinscheduler.plugin.task.datasource.DatasourceUtil.getConnection(DatasourceUtil.java:59)
        at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.executeFuncAndSql(SqlTask.java:181)
        at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.handle(SqlTask.java:152)
        at org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread.run(TaskExecuteThread.java:202)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.codec.binary.Base64
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at org.apache.dolphinscheduler.spi.plugin.DolphinPluginClassLoader.loadClass(DolphinPluginClassLoader.java:81)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 11 common frames omitted

### What you expected to happen

normal run

### How to reproduce

Run a SQL task. The data source uses MySql or others that require a password encoder.

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-task-plugin/dolphinscheduler-task-api/pom.xml
+++ b/dolphinscheduler-task-plugin/dolphinscheduler-task-api/pom.xml
@@ -45,7 +45,10 @@
             <groupId>ch.qos.logback</groupId>
             <artifactId>logback-classic</artifactId>
         </dependency>
-
+        <dependency>
+            <groupId>commons-codec</groupId>
+            <artifactId>commons-codec</artifactId>
+        </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>

